cli/flox/src/commands/services/mod.rs

```diff
@@ -103,6 +110,58 @@ fn processes_by_name_or_default_to_all<'a>(
     }
 }
 
+/// Note that this must be called within an existing activation, otherwise it
+/// will leave behind a process-compose since it doesn't start a watchdog.
+pub async fn start_with_new_process_compose(
+    config: Config,
+    flox: Flox,
+    environment_select: EnvironmentSelect,
+    mut concrete_environment: ConcreteEnvironment,
+    names: &[String],
+) -> Result<Vec<String>> {
+    let environment = concrete_environment.dyn_environment_ref_mut();
+    let lockfile = environment.lockfile(&flox)?;
+    let LockedManifest::Catalog(lockfile) = lockfile else {
+        unreachable!("at least it should be after https://github.com/flox/flox/issues/1858")
+    };
+    for name in names {
+        if !lockfile.manifest.services.contains_key(name) {
+            return Err(service_does_not_exist_error(name));
+        }
+    }
+    Activate {
+        environment: environment_select,
+        // We currently only check for trust for remote environments,
+        // but set this to false in case that changes.
+        trust: false,
+        print_script: false,
+        start_services: true,
+        run_args: vec!["true".to_string()],
+    }
+    .activate(
+        config,
+        flox,
+        concrete_environment,
+        true,
+        &new_services_to_start(names),
+    )
+    .await?;
```

Comment:
what keeps this from exec'ing and not returning back?

Reply:
It passes down to:

- https://github.com/flox/flox/blob/32d01ca33aaff12adeda822422f63b74ba157f4b/cli/flox/src/commands/activate.rs#L501-L516

Worth noting that `start_with_new_process_compose` was only lightly refactored from https://github.com/flox/flox/pull/1913